dkms_prebuild.sh

```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env sh
+
+test -x configure || ./autogen.sh
```

Comment:
do we really need this? users should build from pre configured dist tarballs..

Reply:
Original submitter wanted to support `dkms ldtarball <tarball>` using tar.gz downloaded directly from github, which are not configured. I think it does not hurt in our case. @frejanordsiek, was there any other reason?

Reply:
IMO the standard way is to build from a preconfigured tarball; maybe we can create it as part of the release, as we do for UCX

Reply:
ok removed dkms_prebuild.sh and adapted CI

Reply:
My reasoning was to make it work regardless of whether one used a tarball downloaded from github (or made with `git archive`) or one made from `make dist`. The difference between the two is the dependencies. The former makes autotools a runtime dependency but requires no separate build phase. The latter requires a separate build phase but has less runtime dependencies.

Additionally, in order to actually do `make dist`, you have to actually run `./configure` first. My original motivation was that `./configure` is not guaranteed to succeed automatically without passing arguments and it depends on the kernel actually being supported. Specifically, if one's distro puts the kernel sources in a location xpmem's build scripts don't look in, then the only way for `./configure` to succeed is to explicitly pass the kernel source directory and its version; all just to be able to run `make dist` which then is standalone (dkms passes such things explicitly since it knows where everything is). It also makes it impossible to produce a usable package if one's kernel isn't supported even if the package would later be installed on a machine that does support it. Having it work even from a `git archive` tarball means all it has to do is work in the target environment (no need to work in a builder environment).

Reply:
I see your point, but I think that with #61 merged, we can now always use `./configure --disable-kernel-module` if we only want to obtain a Makefile to run `make dist`.

Reply:
Yes, with #61 , the need for this is a lot less. I did not think about it. Doing it with `--disable-kernel-module` is a good enough solution and a lot less ugly than the hack of `dkms_prebuild.sh`